### ISSUE TITLE
Displays the selection index and the number of matches instead of the function type

### ISSUE THREAD

**Issue by EdmundsEcho** [Shougo/echodoc.vim#42]
I have confirmed that `echodoc` is enabled and `noshowmode` set.  However, the messages show the number of matches and my selection therein (e.g., `match 1 of 4`), then does not display anything once I make a selection.  Any ideas for what might be going on?

I'm using NVIM 0.2.2 Release.
deoplete and neco-ghc do a great job of displaying options
This error presents itself with and without ALE.

**Comment by Shougo** [comment]
Well,....
We have not ESP skills.

You must provide the minimal init.vim and the reproduce instructions.

If not, I will close the issue.

**Comment by EdmundsEcho** [comment]
I understand.  I was hoping you may had heard of such an issue before my needing to do a deeper dive to isolate what was going on.  

Please go ahead and close the issue.  I will try to isolate the issue to provide you a more actionable issue.

**Comment by Shougo** [comment]
> note the green text at the bottom of the image; where I might expect to see the function type signature, I get the deoplete option count

For the green count, you can disable it.

`set shortmess+=c`

If you want to see the function signature, you must insert the completion.

And current echodoc does not support Haskell function signature.
Because, Haskell does not use `()` for functions.

**Comment by EdmundsEcho** [comment]
Thank you.

**Comment by Shougo** [comment]
You can extend echodoc to support haskell.
Like this.

https://github.com/Shougo/echodoc.vim/blob/master/autoload/echodoc/ruby.vim

**Comment by EdmundsEcho** [comment]
Thank you very much.  I wish I had good news.  

1. I disabled the green count using `set shortmess+=c`
1. I loaded the `ruby.vim` file and confirmed my ability to call `echodoc#ruby#get()` from within nvim
1. I have confirmed that `echodoc` is enabled

... to no avail.  

Whether I try using `set cmdheight=2` (or more) or `set noshowmode`, neither display the function types.

![screen shot 2018-03-27 at 9 50 14 am](https://user-images.githubusercontent.com/11245417/37971593-561757f8-31a4-11e8-88d6-81440d416857.png)
.
Note: the image is showing `cmdheight=4` and `noshowmode`.
